h2/src/test/org/h2/test/scripts/TestScript.java

```diff
@@ -427,7 +427,7 @@ private void process(String sql, boolean allowReconnect) throws Exception {
         if (statements != null) {
             statements.add(sql);
         }
-        if (sql.indexOf('?') == -1) {
+        if (sql.indexOf('?') == -1 || sql.indexOf("?'") != -1 || sql.indexOf("?|") != -1 || sql.indexOf("?&") != -1) {
```

Comment:
Such check is not reliable. Characters next to `?` should be tested instead. Not very critical for now, but at least a TODO comment should be placed here to indicate a problem.

Reply:
Current check used to exclude collision between JSON operators and parameter token because it's not obvious how this can be resolved at the parser level.